### PR TITLE
#164069238 Report an article

### PR DESCRIPTION
**Pull request by CryceTruly**
#### What does this PR do?
Adds a feature that enables authenticated users to report inappropriate articles
#### Description of Task to be completed?
add ReportedArticleModel
notify the author of reported abuse

#### How should this be manually tested?
* Clone the repo, cd into Ah-backend-aquaman and check out the branch ft-report-article-164069238
* pip install -r requirements.txt
* python manage.py migrate
* python manage.py runserver
* use postman to hit the endpoint `/articles/<slug>/report/`
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#164069238](https://www.pivotaltracker.com/story/show/164069238)
#### Screenshots (if appropriate) 

`post a report`
Screenshot 2019-03-18 at 20 01 29

`Email sent to the author`
Screenshot 2019-03-18 at 20 05 28

### CODE REVIEW

File: authors/apps/articles/tests/test_report_article.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        Tests method tests a user who is authenticated can report some article
+        """
+
+        self.add_article()
+        self.register_and_login_new_user()
+
+        article = Article.objects.all().first()
+        response = self.client.post(reverse('articles:report-article',
+                                           kwargs={'slug': article.slug}),
+                                    data=valid_reported_article,
```
Comment (wasibani-roy) [inline authors/apps/articles/tests/test_report_article.py]: Please consider reviewing the response for easy reading i suggest kwargs should start at the same level as data

**Review by wasibani-roy — CHANGES_REQUESTED** [review]

File: authors/apps/articles/tests/test_report_article.py (hunk 1 above)
Comment (CryceTruly) [reply authors/apps/articles/tests/test_report_article.py]: I have worked to reformat that, however, the reason why kwargs is on a new line is so as to maintain lines less than 79 characters and since kwargs is still an argument of reverse, it still has to appear like its an argument of reverse for reverse to work on multiple lines.

If we examine where data is, it's essentially saying that even if it's on the third line it has to appear as an argument of post. I guess this is all because of the python way of indenting.

File: authors/apps/articles/admin.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from .models import ReportedArticle
+from django.contrib import admin
+
```
Comment (wasibani-roy) [inline authors/apps/articles/admin.py:3]: Hey @CryceTruly i see you are adding the ReportedArticle model to the admin area of they app why is that necessary??

**Review by BagzieGracious — DISMISSED** [review]

**Review by joelethan — DISMISSED** [review]

**Review by Peace-Apple — DISMISSED** [review]

**Review by wasibani-roy — DISMISSED** [review]

File: authors/apps/core/utils.py [hunk 3]
```diff
@@ -6,13 +6,15 @@ class Utilities:
     """Utility class that contains helper function"""
 
     @staticmethod
-    def email_renderer(data):
+    def send_email(data,intent):
```
Comment (marthamareal) [inline authors/apps/core/utils.py]: @CryceTruly  can you make intent a kwarg with default None? This can reduce on the bugs of where method send_email is used in the codebase

Comment (CryceTruly) [reply authors/apps/core/utils.py]: okay @marthamareal, thanks for that.

File: authors/apps/articles/admin.py (hunk 2 above)
Comment (CryceTruly) [reply authors/apps/articles/admin.py:3]: @wasibani-roy, the whole idea is, when we report, we report to systems administrators and they can access these reported articles in the admin site. That's why i registered the model in the admin.py

**Review by wasibani-roy — DISMISSED** [review]
Good work @CryceTruly all feedback implemented

**Review by Peace-Apple — DISMISSED** [review]

**Review by wasibani-roy — DISMISSED** [review]

**Review by BagzieGracious — DISMISSED** [review]

**Review by Peace-Apple — DISMISSED** [review]

**Review by Peace-Apple — APPROVED** [review]

**Review by BagzieGracious — APPROVED** [review]

**Review by joelethan — APPROVED** [review]
